fendermint/actors/src/manifest.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+use anyhow::{anyhow, Context};
+use cid::Cid;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::CborStore;
+use std::collections::HashMap;
+
+const CHAINMETADATA_ACTOR_NAME: &str = "chainmetadata";
+
+/// A mapping of internal actor CIDs to their respective types.
+pub struct Manifest {
+    pub chainmetadata_code: Cid,
```

Comment:
Is it necessary to separate the builtin-actor manifest from the custom actor manifest? I don't see `is_chainmetadata_actor` being used anywhere, but I wonder if there is a way to have a unified approach?

Reply:
I see that the original `Manifest` assigns `code_id` based on this sequence so if we added more entries to the root CID somehow, they could be unstable if the builtin actors are expanded by them in subsequent versions.

What I'm thinking is maybe we don't need a known code ID for this actor. I'm not even sure what the usecase for it is, but for example the IPC actors don't have it, just the identity IDs, which is enough to send messages to a well-known location. Then we could put our own actors at the end of a common list and use a single manifest. 

It's a bit strange anyway that actors have a name and a CID, and then they get an arbitrary ID based on their appearance, which also gets hardcoded in the code as a stable `_CODE_ID` yet if we decided to delete an actor from the middle of our list it would all go haywire. Why not just use the name if we have to look up its CID?

Perhaps this is a pattern we don't have to perpetuate?